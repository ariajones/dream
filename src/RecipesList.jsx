import React from 'react'

import RecipeListItem from './RecipeListItem'

function RecipesList() {
  return (
    <div className="RecipesList">
      <h2>RecipesList</h2>
      <RecipeListItem />
    </div>
  )
}

export default RecipesList