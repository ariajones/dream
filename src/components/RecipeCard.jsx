import React from "react"
import { Link } from "react-router-dom"
import { deleteCardRecipe } from '../utils'

function RecipeCard ({ recipe }) {
    return (
      <>
      <div className="card">
        <Link to={{ pathname: `/recipe/${recipe.id}`, recipe: recipe}}> {/* --- DYNAMIC RECIPE ID PATH--- */}
            <div className="card-image">
              <figure className="image is-5by1">
                <img src={recipe.imagePath} alt={recipe.name}/> {/* --- IMAGE PATH --- */}
              </figure>
            </div>
              <div className="card-content">
                <div className="media mb-0">
                  <div className="media-content">
                    <p className="title is-5 mb-0">{recipe.name}</p> {/* --- NAME OF RECIPE --- */}
                    <div className="field is-grouped">
                      <p className="control">
                        <Link to={`/recipe/${recipe.id}`}><button className="button is-link is-small is-rounded">
                          <i className="far fa-calendar-plus"></i>
                        </button>
                        </Link>
                      </p>
                      <p className="control">
                        <Link to={`/recipe/edit/${recipe.id}`}><button className="button is-small is-rounded is-primary">
                        <i class="far fa-edit"></i>
                        </button>
                        </Link>
                      </p>
                      <p className="control">
                      <Link to='/recipes'>
                      <button onClick={() => deleteCardRecipe(recipe)}className="button is-danger is-small is-rounded">
                        <i className="fas fa-trash-alt"></i>
                        </button>
                      </Link>
                      </p>
                    </div>
                  </div>
                </div>

                <div className="content">
                  <p className="mb-0">Serves: {recipe.serves}</p> {/* --- SERVES --- */}
                  <p>Prep time: {recipe.prepTime}</p> {/* --- PREP TIME --- */}
                </div>
            </div>
        </Link>
      </div>
            
      <div className="seperation"></div>
    </>
  )
}

export default RecipeCard