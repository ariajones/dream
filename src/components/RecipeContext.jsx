import React, { useState, createContext, useEffect } from 'react'
import firebase from 'firebase/app'

export const RecipeContext = createContext()

export const RecipeProvider = ({ children }) => {
  const [recipes, setRecipes] = useState([])

  useEffect(() => {
    const unsubscribe = firebase
      .firestore()
      .collection('recipes')
      .onSnapshot(snapshot => {
        const newRecipes = snapshot.docs.map(doc => ({
          id: doc.id,
          ...doc.data()
        }))
        setRecipes(newRecipes)
      })

    return () => unsubscribe()
  }, [])

  return (
    <RecipeContext.Provider value={[recipes, setRecipes]}>
      {children}
    </RecipeContext.Provider>
  )
}
