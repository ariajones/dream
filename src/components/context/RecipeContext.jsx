import React, { useState, createContext, useEffect, useContext } from 'react'
import firebase from 'firebase/app'
import 'firebase/auth'

import { UserContext } from './UserContext'

export const RecipeContext = createContext()

export const RecipeProvider = ({ children }) => {
  const {user} = useContext(UserContext)
  const [recipes, setRecipes] = useState([])

  useEffect(() => {
    const unsubscribe = firebase 
      .firestore()
      .collection('recipes')
      .where('userId', '==', user ? user.uid : null)
      .onSnapshot(snapshot => {
        const newRecipes = snapshot.docs.map(doc => ({
          id: doc.id,
          ...doc.data()
        }))
        setRecipes(newRecipes)
      })

    return () => unsubscribe()
  }, [user])

  return (
    <RecipeContext.Provider value={[recipes, setRecipes]}>
      {children}
    </RecipeContext.Provider>
  )
}