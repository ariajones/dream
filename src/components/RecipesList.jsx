import React, { useContext, useState } from 'react'

import { RecipeContext } from './context/RecipeContext'
import RecipeCard from './RecipeCard'
import Seed from './Seed'

import { sortRecipes } from '../utils.js'

function RecipesList () {
  const [recipes] = useContext(RecipeContext)
  const [sortBy, setSortBy] = useState('NAME_ASC')
  const sortedRecipes = sortRecipes([...recipes], sortBy)

  return (
    <>
      <div className="RecipesList">
        <Seed /> {/* --- TO BE DELETED FROM PRODUCTION --- */}
        <div className='recipeSort'>
          <label>Sort By:</label>{' '}
          <select value={sortBy} onChange={evt => setSortBy(evt.target.value)}>
            <option value='NAME_ASC'>Name (A-Z)</option>
            <option value='NAME_DESC'>Name (Z-A)</option>
          </select>
        </div>
        <ul>
          {sortedRecipes.map(recipe =>
            <li key={recipe.id}>
              <RecipeCard recipe={recipe} />
            </li>
          )}
        </ul>
      </div>
    </>
  )
}

export default RecipesList
