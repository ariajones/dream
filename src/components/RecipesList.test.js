import React, { useContext, useState } from 'react'
import renderer from 'react-test-renderer'
import { HashRouter, HashRouter as Router } from 'react-router-dom'

import RecipesList from './RecipesList'
import { RecipeContext } from './RecipeContext'
import RecipeCard from './RecipeCard'

export const FakeRecipeProvider = ({ children }) => {
    const [recipes, setRecipes] = useState([{
        id: "MOqgTKsRp9OwepdSupbr",
    method: ["Preheat the oven to 200ºC/400ºF/gas 6. Peel 1 oni…"],
    serves: "4",
    prepTime: "40 Minutes - Not too tricky",
    name: "Roasted black bean burgers",
    ingredients: ["1½ red onions", "100 g rye bread", "ground corian…"],
    imagePath: "https://imagesvc.meredithcorp.io/v3/mm/image?url=https%3A%2F%2Fimages.media-allrecipes.com%2Fuserphotos%2F5858755.jpg"
    }])
  
    return (
        <RecipeContext.Provider value={[recipes, setRecipes]}>
          {children}
        </RecipeContext.Provider>
      )

}


it ('renders component correctly', () => {
    const dream = renderer
    .create(
        <HashRouter>
            <FakeRecipeProvider>
                <RecipesList>
                    <RecipeCard/>
                </RecipesList>
            </FakeRecipeProvider>
        </HashRouter>)
    .toJSON()
expect(dream).toMatchSnapshot()
})