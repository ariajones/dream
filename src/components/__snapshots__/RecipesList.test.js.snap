// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component correctly 1`] = `
<div
  className="RecipesList"
>
  <div>
    <button
      onClick={[Function]}
    >
      DELETE ALL RECIPES
    </button>
    <button
      onClick={[Function]}
    >
      SEED RECIPES
    </button>
  </div>
   
  <div
    className="recipeSort"
  >
    <div
      className="list"
    >
      <input
        onChange={[Function]}
        placeholder="search for a recipe..."
        type="text"
      />
      <label>
        Sort By:
      </label>
       
      <select
        onChange={[Function]}
        value="NAME_ASC"
      >
        <option
          value="NAME_ASC"
        >
          Name (A-Z)
        </option>
        <option
          value="NAME_DESC"
        >
          Name (Z-A)
        </option>
        <option
          value="SERVES_ASC"
        >
          Serves (Ascending)
        </option>
        <option
          value="SERVES_DESC"
        >
          Serves (Descending)
        </option>
      </select>
    </div>
  </div>
  <ul>
    <li>
      <div
        className="card"
      >
        <a
          href="#/recipe/MOqgTKsRp9OwepdSupbr"
          onClick={[Function]}
        >
           
          <div
            className="card-image"
          >
            <div
              className="container"
            >
              <figure
                className="image is-1by10"
              >
                <img
                  alt="Roasted black bean burgers"
                  src="https://imagesvc.meredithcorp.io/v3/mm/image?url=https%3A%2F%2Fimages.media-allrecipes.com%2Fuserphotos%2F5858755.jpg"
                />
                <div
                  className="top-right"
                >
                  <div
                    className="control"
                  >
                    <i
                      className="fas fa-plus"
                    />
                    <a
                      href="#/recipe/MOqgTKsRp9OwepdSupbr"
                      onClick={[Function]}
                    />
                  </div>
                </div>
                 
              </figure>
            </div>
          </div>
          <div
            className="card-content"
          >
            <div
              className="media mb-0"
            >
              <div
                className="media-content"
              >
                <div
                  className="subtitle is-5 mb-0"
                >
                  Roasted black bean burgers
                </div>
                 
              </div>
            </div>
            <div
              className="content"
            >
              <div
                className="subtitle is-6 mb-0"
              >
                Serves: 
                4
              </div>
               
              <div
                className="subtitle is-6"
              >
                Prep time: 
                40 Minutes - Not too tricky
              </div>
               
            </div>
          </div>
        </a>
      </div>
      <div
        className="seperation"
      />
    </li>
  </ul>
</div>
`;
